src/library/export/trackexport.cpp

```diff
@@ -0,0 +1,129 @@
+#include "library/export/trackexport.h"
+
+#include <QFileInfo>
+#include <QMessageBox>
+#include <QDebug>
+
+void TrackExport::run() {
+    int i = 0;
+    for (const auto& track : m_tracks) {
+        QString path = track->getLocation();
+        QFileInfo fileinfo(path);
+        emit(progress(fileinfo.fileName(), i, m_tracks.size()));
+        exportTrack(track);
+        if (m_bStop) {
+            emit(canceled());
+            return;
+        }
+        ++i;
+        emit(progress(fileinfo.fileName(), i, m_tracks.size()));
```

Comment:
yeah it's a little hacky, but it ensures that we emit the progress bar state before we begin copying and after we finish.   It also means the progress updates in a sane way:

File 1
[       ]

File 1
[-     ]

File 2
[-    ]

File 2
[--   ]